src/Sentry/TransactionTracer.cs

```diff
@@ -6,9 +6,11 @@ namespace Sentry;
 /// <summary>
 /// Transaction tracer.
 /// </summary>
-public class TransactionTracer : ITransaction, IHasDistribution, IHasTransactionNameSource, IHasMeasurements
+public class TransactionTracer : ITransaction, IHasDistribution, IHasTransactionNameSource, IHasMeasurements, IDisposable
```

Comment:
Is adding IDisposable is a "breaking change" as we expect anyone bumping the SDK to now deal with disposing this. How do we expect the caller to deal with this? 
IIRC we return this as `ITransaction` in places, right? We'd need to document/expect users to try cast to IDisposable and call it if not null.

Also, Dispose could only be called after `Finish` which defeats the purpose of idle transaction (which is completed from the background thread)

Reply:
> Is adding IDisposable is a "breaking change" as we expect anyone bumping the SDK to now deal with disposing this. How do we expect the caller to deal with this?
> IIRC we return this as `ITransaction` in places, right? We'd need to document/expect users to try cast to IDisposable and call it if not null.

You're correct. It would be considered a breaking change.
`IDisposable` is there to dispose of the `Timer`. But I wonder, since we control the `Finish` and stop the timer there, the timer could be disposed of w/o the need for the user to do anything. I.e. dispose of the timer in the `Finish` and remove the breaking `IDisposable` on the `TransactionTracer`, removing the need for the user to handle the disposing.